### ISSUE TITLE
Feature Request: Support ECS Service Health Check Grace Period

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#2786]
Amazon Elastic Container Service (Amazon ECS) service scheduler now allows you to define a grace period to prevent premature shutdown of newly instantiated tasks: https://aws.amazon.com/about-aws/whats-new/2017/12/amazon-ecs-adds-elb-health-check-grace-period/

Prerequisite: aws-sdk-go v.1.12.53 (already merged)

### Terraform Version

terraform 0.10+
terraform-provider-aws 1.6.0

### Affected Resource(s)

- `aws_ecs_service`

### Terraform Configuration Files

```hcl
resource "aws_ecs_service" "example" {
  # new configuration
  health_check_grace_period_seconds = 300
  # existing configuration
}
```

### Expected Behavior

`HealthCheckGracePeriodSeconds` integer added to `CreateServiceInput`/`UpdateServiceInput` when `health_check_grace_period_seconds` optionally set in Terraform configuration. It has a max value of 1800.

### Actual Behavior

New feature.

### References

* https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html
* https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#CreateServiceInput

**Comment by rafilkmp3** [comment]
if its merged why i can use ?? 
i facing with
: invalid or unknown key: health_check_grace_period_seconds

**Comment by bflad** [comment]
Hi @rafilkmp3 - sorry for any confusion. This will be in version 1.7.0 of the provider, which has not been fully released yet, although it likely will be soon. You can check the code repository CHANGELOG here to see when its released: https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md

If you need the functionality immediately, you can grab the master branch of the source code to build and use the updated provider yourself right now.

**Comment by rafilkmp3** [comment]
I did this, but after build i don't now get a binary like terraform downloads into .terraform/plugins/linux_amd64 , 
how i can use the builded plugin ?

**Comment by bflad** [comment]
After running `make build` the provider binary should be dropped as `${GOPATH}/bin/terraform-provider-aws`, which can then be moved/copied as needed.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
